ocamlfind error when using `core`

When adding `core` to `ocamlfind-dependencies` the compiliation of an empty file fails with:

```
ninja: Entering directory `lib/bs/bytecode'
[2/2] Building src/Index.mlast.d
[1/2] Building src/Index.cmo
FAILED: src/Index.cmo src/Index.cmi 
ocamlfind ocamlc    -I /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml -I src -package core  -passopt -thread -passopt -color -passopt always -o src/Index.cmo -w -30-40+6+7+27+32..39+44+45+101 -g -c -intf-suffix .mliast_simple -impl src/Index.mlast_simple 
findlib: [WARNING] Interface int64.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface listLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface unixLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface lexing.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface scanf.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface dynlink.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface gc.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface callback.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface nativeint.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface arrayLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface queue.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface big_int.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface num.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface genlex.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface digest.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface printexc.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface sort.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface int32.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface map.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface stdLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface printf.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface camlinternalMod.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface camlinternalLazy.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface list.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface stream.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface parsing.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface camlinternalFormatBasics.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface char.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface oo.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface stack.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface hashtbl.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface set.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface nat.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface filename.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface bigarray.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface ratio.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface random.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface std_exit.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface lazy.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface camlinternalFormat.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface arith_status.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface obj.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface complex.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface format.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface unix.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface camlinternalOO.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface stringLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface moreLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface marshal.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface buffer.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface arg.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface sys.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface bytesLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface weak.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface bytes.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface string.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface str.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface profiling.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface pervasives.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
findlib: [WARNING] Interface array.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.02.3+buckle-master/lib/ocaml
ocamlfind: Error from package `threads': Missing -thread or -vmthread switch
ninja: build stopped: subcommand failed.
```

I'm not sure how importing the warnings are, the line before the last one seems too be the crux:

`ocamlfind: Error from package `threads': Missing -thread or -vmthread switch`

Which is weird since `-thread` is added by default since https://github.com/bsansouci/bsb-native/pull/8/files

Also adding it to `ocaml-flags` does not help.

Here is an example repo: https://github.com/despairblue/curly-pancake

@bsansouci I updated the test repo to master, removed node_modules, reinstalled them cleanly and I'm still getting this error:

```
ninja: Entering directory `lib/bs/bytecode'
[1/2] Building src/Index.cmo
FAILED: src/Index.cmo src/Index.cmi 
ocamlfind ocamlc    -I /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml -I src -package core  -passopt -thread -passopt -color -passopt always -o src/Index.cmo -w -30-40+6+7+27+32..39+44+45+101 -g -c -intf-suffix .mliast_simple -impl src/Index.mlast_simple 
findlib: [WARNING] Interface int64.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface listLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface callback.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface sort.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface int32.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface printf.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface camlinternalMod.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface camlinternalFormatBasics.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface bigarray.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface random.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface std_exit.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface obj.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface complex.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface stringLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface moreLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface buffer.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface bytesLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface profiling.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface dynlink.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface gc.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface queue.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface big_int.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface stream.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface parsing.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface oo.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface nat.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface arith_status.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface format.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface unix.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface camlinternalOO.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface sys.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface string.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface str.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface pervasives.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface scanf.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface printexc.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface map.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface list.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface filename.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface ratio.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface lazy.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface arg.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface bytes.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface unixLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface lexing.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface nativeint.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface arrayLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface num.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface genlex.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface digest.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface stdLabels.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface camlinternalLazy.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface char.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface stack.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface hashtbl.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface set.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface camlinternalFormat.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface marshal.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface weak.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
findlib: [WARNING] Interface array.cmi occurs in several directories: /home/despairblue/git/wilder-native/node_modules/bs-platform/vendor/ocaml/lib/ocaml, /home/despairblue/.opam/4.06.0/lib/ocaml
ocamlfind: Error from package `threads': Missing -thread or -vmthread switch
ninja: build stopped: subcommand failed.
```

--- a/jscomp/bsb/bsb_default.ml
+++ b/jscomp/bsb/bsb_default.ml
@@ -52,10 +52,9 @@ let bsc_flags =
     "-color"; "always" 
   ]
 
-let ocaml_flags = ["-no-alias-deps"; "-thread"; "-I"; "+compiler-libs"]
+let ocaml_flags = ["-no-alias-deps"]
 
 let refmt_flags = ["--print"; "binary"]
-
 
 let main_entries = [Bsb_config_types.JsTarget "Index"]
 

--- a/jscomp/bsb/bsb_ninja_gen.ml
+++ b/jscomp/bsb/bsb_ninja_gen.ml
@@ -106,19 +106,25 @@ let output_ninja_and_namespace_map
   
   let use_ocamlfind = ocamlfind_dependencies <> [] in
   
-  let ocaml_flags = (List.fold_left (fun acc  v -> 
-    match v with
-    | "compiler-libs" -> "-I" :: "+compiler-libs" :: acc
-    | "threads" -> "-thread" :: acc
-    | _ -> acc
-  ) [] ocaml_dependencies) in
   let external_ocaml_dependencies = List.fold_left (fun acc v -> Depend.StringSet.add v acc) external_ocaml_dependencies ocaml_dependencies in
   let external_ocaml_dependencies = Depend.StringSet.elements external_ocaml_dependencies in
-  
+
   let ocaml_flags =
     Bsb_build_util.flag_concat
       (if use_ocamlfind then "-passopt" else Ext_string.single_space)
       (ocaml_flags @ ["-color"; "always"])  in
+
+
+  let ocaml_flags = (List.fold_left (fun acc v ->
+    match v with
+    | "compiler-libs" ->
+      (if use_ocamlfind then
+        "-package +compiler-libs.common " else
+        "-I +compiler-lib ") ^ acc
+    | "threads" -> "-thread " ^ acc
+    | _ -> acc
+  ) ocaml_flags ocaml_dependencies) in
+
   let bs_super_errors = if main_bs_super_errors && not use_ocamlfind && backend != Bsb_config_types.Js then "-bs-super-errors" else "" in
   
   let oc = open_out_bin (cwd // Bsb_config.lib_bs // nested // Literals.build_ninja) in
@@ -247,13 +253,16 @@ let output_ninja_and_namespace_map
   in 
   let ocaml_lib = Bsb_build_util.get_ocaml_lib_dir ~is_js:(backend = Bsb_config_types.Js) root_project_dir in
   let emit_bsc_lib_includes source_dirs = 
+    let common_include_flags =
+      (all_includes (if namespace = None then source_dirs
+      else Filename.current_dir_name :: source_dirs)) in
+    let post_ocamlfind_include_flags =
+      (if use_ocamlfind then common_include_flags else
+      (ocaml_lib :: common_include_flags)) in
     Bsb_ninja_util.output_kv
-      Bsb_build_schemas.bsc_lib_includes 
-      (Bsb_build_util.flag_concat dash_i @@ 
-       (ocaml_lib :: (all_includes 
-                 (if namespace = None then source_dirs 
-                  else Filename.current_dir_name :: source_dirs) )))  oc 
-  in   
+      Bsb_build_schemas.bsc_lib_includes
+       (Bsb_build_util.flag_concat dash_i @@ post_ocamlfind_include_flags) oc
+  in
   let  bs_groups, bsc_lib_dirs, static_resources =
     let number_of_dev_groups = Bsb_dir_index.get_current_number_of_dev_groups () in
     if number_of_dev_groups = 0 then

--- a/lib/bsb.ml
+++ b/lib/bsb.ml
@@ -10438,10 +10438,9 @@ let bsc_flags =
     "-color"; "always" 
   ]
 
-let ocaml_flags = ["-no-alias-deps"; "-thread"; "-I"; "+compiler-libs"]
+let ocaml_flags = ["-no-alias-deps"]
 
 let refmt_flags = ["--print"; "binary"]
-
 
 let main_entries = [Bsb_config_types.JsTarget "Index"]
 
@@ -17518,19 +17517,25 @@ let output_ninja_and_namespace_map
   
   let use_ocamlfind = ocamlfind_dependencies <> [] in
   
-  let ocaml_flags = (List.fold_left (fun acc  v -> 
-    match v with
-    | "compiler-libs" -> "-I" :: "+compiler-libs" :: acc
-    | "threads" -> "-thread" :: acc
-    | _ -> acc
-  ) [] ocaml_dependencies) in
   let external_ocaml_dependencies = List.fold_left (fun acc v -> Depend.StringSet.add v acc) external_ocaml_dependencies ocaml_dependencies in
   let external_ocaml_dependencies = Depend.StringSet.elements external_ocaml_dependencies in
-  
+
   let ocaml_flags =
     Bsb_build_util.flag_concat
       (if use_ocamlfind then "-passopt" else Ext_string.single_space)
       (ocaml_flags @ ["-color"; "always"])  in
+
+
+  let ocaml_flags = (List.fold_left (fun acc v ->
+    match v with
+    | "compiler-libs" ->
+      (if use_ocamlfind then
+        "-package +compiler-libs.common " else
+        "-I +compiler-lib ") ^ acc
+    | "threads" -> "-thread " ^ acc
+    | _ -> acc
+  ) ocaml_flags ocaml_dependencies) in
+
   let bs_super_errors = if main_bs_super_errors && not use_ocamlfind && backend != Bsb_config_types.Js then "-bs-super-errors" else "" in
   
   let oc = open_out_bin (cwd // Bsb_config.lib_bs // nested // Literals.build_ninja) in
@@ -17659,13 +17664,16 @@ let output_ninja_and_namespace_map
   in 
   let ocaml_lib = Bsb_build_util.get_ocaml_lib_dir ~is_js:(backend = Bsb_config_types.Js) root_project_dir in
   let emit_bsc_lib_includes source_dirs = 
+    let common_include_flags =
+      (all_includes (if namespace = None then source_dirs
+      else Filename.current_dir_name :: source_dirs)) in
+    let post_ocamlfind_include_flags =
+      (if use_ocamlfind then common_include_flags else
+      (ocaml_lib :: common_include_flags)) in
     Bsb_ninja_util.output_kv
-      Bsb_build_schemas.bsc_lib_includes 
-      (Bsb_build_util.flag_concat dash_i @@ 
-       (ocaml_lib :: (all_includes 
-                 (if namespace = None then source_dirs 
-                  else Filename.current_dir_name :: source_dirs) )))  oc 
-  in   
+      Bsb_build_schemas.bsc_lib_includes
+       (Bsb_build_util.flag_concat dash_i @@ post_ocamlfind_include_flags) oc
+  in
   let  bs_groups, bsc_lib_dirs, static_resources =
     let number_of_dev_groups = Bsb_dir_index.get_current_number_of_dev_groups () in
     if number_of_dev_groups = 0 then

